pkg/cli/manifest/request.go

```diff
@@ -0,0 +1,603 @@
+package manifest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/r3labs/sse"
+
+	"github.com/seal-io/walrus/pkg/cli/api"
+	"github.com/seal-io/walrus/pkg/cli/config"
+	"github.com/seal-io/walrus/pkg/dao/types/status"
+	"github.com/seal-io/walrus/utils/gopool"
+	"github.com/seal-io/walrus/utils/json"
+	"github.com/seal-io/walrus/utils/log"
+	"github.com/seal-io/walrus/utils/pointer"
+	"github.com/seal-io/walrus/utils/strs"
+)
+
+const (
+	operationWatch       = "list"
+	operationBatchCreate = "batch-create"
+	operationBatchDelete = "batch-delete"
+	operationGet         = "get"
+	operationPatch       = "patch"
+)
+
+type ObjectStatus uint
+
+const (
+	statusNotFound ObjectStatus = iota
+	statusUnchanged
+	statusChanged
+)
+
+type result struct {
+	err  error
+	obj  Object
+	objs []Object
+}
+
+// PatchObjects send patches objects request.
+func PatchObjects(sc *config.Config, group string, objs ObjectByScope) (success, failed ObjectSet, err error) {
+	if len(objs) == 0 {
+		return
+	}
+
+	patchOpt := api.OpenAPI.GetOperation(group, operationPatch)
+	if patchOpt == nil {
+		return success, failed, fmt.Errorf("not found patch operation for %s", group)
+	}
+
+	var (
+		results = make(chan result, 1)
+		all     = objs.All()
+	)
+
+	for _, obj := range all {
```

Comment:
As our server has rate limit when object number is large will rate limit cause some patch failed?

Reply:
Now it will retry three times.